### PR TITLE
Change selects dialogs to autocompletes #643

### LINKED ISSUE #643: Change all dialog select components to auto completes
- System importance
- Obsolete dialog (is obsolete) 
- catalogue category property migration (add, edit)

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

changes last few selects to autocompletes

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #643

### CODE REVIEW

File: cypress/e2e/with_mock_data/items.cy.ts [hunk 1]
```diff
@@ -358,21 +358,16 @@ describe('Items', () => {
...
     cy.findByLabelText('Resolution (megapixels) *').clear();
     cy.findByLabelText('Sensor Type *').clear();
-    cy.findByLabelText('Broken *').clear();
 
     cy.findByRole('button', { name: 'Next' }).click();
 
     cy.findAllByText(
       'Please enter a valid value as this field is mandatory'
     ).should('have.length', 2);
-    cy.findByText('Please select either True or False').should('exist');
```
Comment (joelvdavies) [inline cypress/e2e/with_mock_data/items.cy.ts:368]: Why was this removed? Shouldn't the errors still be there?

File: src/catalogue/items/catalogueItemsDialog.component.test.tsx [hunk 2]
```diff
@@ -846,17 +846,10 @@ describe('Catalogue Items Dialog', () => {
       expect(costHelperText).toBeInTheDocument();
       expect(daysToReplaceHelperText).toBeInTheDocument();
 
-      expect(
-        screen.getByText(
-          'Please choose a manufacturer, or add a new manufacturer'
-        )
-      ).toBeInTheDocument();
```
Comment (joelvdavies) [inline src/catalogue/items/catalogueItemsDialog.component.test.tsx:853]: Why has this been removed?

File: src/catalogue/items/catalogueItemsDialog.component.test.tsx [hunk 3]
```diff
@@ -876,12 +868,6 @@ describe('Catalogue Items Dialog', () => {
         'Please enter a valid value as this field is mandatory'
       );
 
-      const mandatoryFieldBooleanHelperText = screen.getByText(
-        'Please select either True or False'
-      );
-
-      expect(mandatoryFieldBooleanHelperText).toBeInTheDocument();
```
Comment (joelvdavies) [inline src/catalogue/items/catalogueItemsDialog.component.test.tsx:883]: Same here

**Review by joelvdavies — CHANGES_REQUESTED** [review]

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]
image

image

can you also revert the minimum width and width back so it doesn;t trigger the scroll bar

File: src/catalogue/items/catalogueItemsDialog.component.test.tsx (hunk 2 above)
Comment (MatteoGuarnaccia5) [reply src/catalogue/items/catalogueItemsDialog.component.test.tsx:853]: Same reason as above - it is no longer clearable

File: src/catalogue/items/catalogueItemsDialog.component.test.tsx (hunk 3 above)
Comment (MatteoGuarnaccia5) [reply src/catalogue/items/catalogueItemsDialog.component.test.tsx:883]: Same reason as above - it is no longer clearable

File: src/items/itemDialog.component.test.tsx [hunk 4]
```diff
@@ -860,8 +852,8 @@ describe('ItemDialog', () => {
 
       await user.click(screen.getByRole('button', { name: 'Next' }));
 
-      const pumpingSpeedAutoComplete = screen.getAllByRole('combobox')[0];
-      await user.type(pumpingSpeedAutoComplete, '{delete}');
+      const ultimatePressureTextBox = screen.getAllByRole('textbox')[0];
```
[the remark this answers is not in the document]
Comment (MatteoGuarnaccia5) [reply src/items/itemDialog.component.test.tsx:855]: Same reason as above - it is no longer clearable, so to test that it does error on unfilled fields I cleared a different one

File: src/catalogue/items/catalogueItemsDialog.component.tsx [hunk 5]
```diff
@@ -850,11 +851,19 @@ function CatalogueItemsDialog(props: CatalogueItemsDialogProps) {
...
                                 )}`}
-                                value={(propertyValues[index] as string) ?? ''}
+                                value={
+                                  propertyValues[index]
+                                    ? (propertyValues[index] as string)
+                                        .charAt(0)
+                                        .toUpperCase() +
+                                      (propertyValues[index] as string).slice(1)
+                                    : ''
+                                }
```
[the remark this answers is not in the document]
Comment (MatteoGuarnaccia5) [reply src/catalogue/items/catalogueItemsDialog.component.tsx:866]: image

This causes the value in the field to be different from the label, I can make the labels lowercase so that they match or I can leave as is

File: src/items/itemDialog.component.tsx [hunk 6]
```diff
@@ -886,11 +888,19 @@ function ItemDialog(props: ItemDialogProps) {
...
                                   '-'
                                 )}`}
-                                value={(propertyValues[index] as string) ?? ''}
+                                value={
+                                  propertyValues[index]
+                                    ? (propertyValues[index] as string)
+                                        .charAt(0)
+                                        .toUpperCase() +
+                                      (propertyValues[index] as string).slice(1)
+                                    : null
```
[the remark this answers is not in the document]
Comment (MatteoGuarnaccia5) [reply src/items/itemDialog.component.tsx:902]: same thing with the values and labels not matching

[the remark this answers is not in the document]
Comment (joshuadkitenge) [reply src/items/itemDialog.component.tsx:902]: the values and label for the properties should keep the same case sensitivity that the user entering in
 with

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]
works well just a few minor comments 


image
image

**Can you center the value please**



image
**Can you make this True and False in the migration dialog**

**Review by joelvdavies — APPROVED** [review]
